cpp/src/knn/knn_opg_common.cu

```diff
@@ -532,38 +642,38 @@ void opg_knn(raft::handle_t &handle, std::vector<Matrix::Data<T> *> *out,
                           handle.get_device_allocator(), cur_batch_size, k,
                           cur_query_ptr, rowMajorIndex, rowMajorQuery);
 
-        // Synchronize before running labels copy
-        CUDA_CHECK(cudaStreamSynchronize(stream));
-
         copy_outputs(res.data(), res_I.data(), y, (size_t)cur_batch_size,
```

Comment:
Once the local KNN has run and the indices were locally merged we need to get the associated outputs/labels from the index partitions. The objective of the `copy_outputs` function is to copy the labels/outputs contained in the index partitions knowing the locally merged indices. What name do you prefer for this operation?

Reply:
Can you add a doxygen comment to `copy_outputs`? 

> What name do you prefer for this operation?

Maybe something like `copy_label_outputs_from_index_parts`? I'm more partial to the longer name in this case because there's so many different types of inputs and outputs in this algorithm.